### DISCUSSION TITLE
custom `boxShadow` being deleted from `className` when color is specified

### DISCUSSION THREAD

**Discussion by hexiro** [dcastil/tailwind-merge#275] [Help] [answered]
Hello! 

When using a custom `boxShadow` defined in the tailwind config, twMerge will delete the `shadow-[name]` from the `className` without any other shadow overriding it if a custom color is specified. This could be an issue with my configuration. I'm using the out-the-box `twMerge` and don't know if I need to do anything to inform `twMerge` about my tailwind config. 

Any help would be appreciated!

ex.
```ts
className={twMerge("shadow-custom shadow-red-500")}
```
emits `shadow-red-500` as the final class name
![image](https://github.com/dcastil/tailwind-merge/assets/42787085/1cd248fa-d173-433a-b7da-c45b591f8d7d)

**Comment by dcastil** [comment, marked as answer]
Hey @hexiro! 👋

tailwind-merge doesn't have access to the tailwind.config.js file and you need to configure it separately. The default out-of-the-box config of tailwind-merge assumes all unknown classes which could be a color (like `shadow-custom`) are a color, this is why it de-duplicated it when you use `shadow-red-500` afterwards.

Here is an example on how to configure tailwind-merge: https://github.com/dcastil/tailwind-merge/blob/v1.14.0/docs/recipes.md#adding-custom-scale-from-tailwind-config-to-tailwind-merge-config.

And here is the documentation on how the tailwind-merge configuration works: https://github.com/dcastil/tailwind-merge/blob/v1.14.0/docs/configuration.md#usage-with-custom-tailwind-config.

Related: https://github.com/dcastil/tailwind-merge/issues/274, https://github.com/dcastil/tailwind-merge/issues/250, https://github.com/dcastil/tailwind-merge/issues/207

**Reply by hexiro** [reply]
Ah, I see! 

Since making this post I've learned about [ring](https://tailwindcss.com/docs/ring-width) in tailwind which solved my issue in this example, but If I run into this issue again I'll be sure to try out configuring tailwind-merge.

Thank you for the response and thank you for making this package !!!
